core/src/test/java/org/elasticsearch/index/mapper/date/DateBackwardsCompatibilityTests.java

```diff
@@ -0,0 +1,208 @@
+
+        XContentBuilder document = XContentFactory.jsonBuilder()
+                .startObject()
+                .field("date_field", randomBoolean() ? String.valueOf(date) : date)
+                .endObject();
+        index(document);
+
+        // no results in expected time range
+        QueryBuilder query = QueryBuilders.rangeQuery("date_field").from("2015-06-23").to("2015-06-24").format("dateOptionalTime");
+        SearchResponse response = client().prepareSearch(index).setQuery(query).get();
+        assertNoSearchHits(response);
+
+        // result in unix timestamp range
+        QueryBuilder timestampQuery = QueryBuilders.rangeQuery("date_field").from(2015062300000L).to(2015062302000L);
+        assertHitCount(client().prepareSearch(index).setQuery(timestampQuery).get(), 1);
+
+        // result should also work with regular specified dates
+        QueryBuilder regularTimeQuery = QueryBuilders.rangeQuery("date_field").from("2033-11-08").to("2033-11-09").format("dateOptionalTime");
+        assertHitCount(client().prepareSearch(index).setQuery(regularTimeQuery).get(), 1);
+    }
+
+    public void testThatPost2xIndicesNumbersAreTreatedAsStrings() throws Exception {
+        // looks like a unix time stamp but is meant as 2016-06-23T01:00:00.000 - see the specified date format
+        long date = 2015062301000l;
+
+        XContentBuilder mapping = jsonBuilder().startObject().startObject("properties")
+                .startObject("date_field").field("type", "date").field("format","yyyyMMddHHSSS").endObject()
+                .endObject().endObject();
+        index = createIndex(Version.CURRENT, mapping);
+
+        XContentBuilder document = XContentFactory.jsonBuilder()
+                .startObject()
+                .field("date_field", randomBoolean() ? String.valueOf(date) : date)
+                .endObject();
+        index(document);
+
+        // search for date in time range
+        QueryBuilder query = QueryBuilders.rangeQuery("date_field").from("2015-06-23").to("2015-06-24").format("dateOptionalTime");
+        SearchResponse response = client().prepareSearch(index).setQuery(query).get();
+        assertHitCount(response, 1);
+    }
+
+    @Test(expected = MapperParsingException.class)
+    public void testDynamicDateDetectionIn2xDoesNotSupportEpochs() throws Exception {
+        XContentBuilder mapping = jsonBuilder().startObject()
+                .startArray("dynamic_date_formats").value("dateOptionalTime").value("epoch_seconds").endArray()
+                .endObject();
+        createIndex(Version.CURRENT, mapping);
+    }
+
+    /* 2.0
+     * For dynamic date detection:
+     *   only check string values (don't coerce numerics)
+     *   accept any formats except epoch_ms and epoch_seconds
+     *   mapping should add just the matching format (optionally append epoch_ms?)
+     */
+    // TODO add a test for dynamic date detection on any strings, but keep performance in mind
+
+
+    /* TODO: DECIDE HERE
```

Comment:
can this comment block be removed or moved to DateFieldMapper? seems like the wrong place for api/impl notes